stlab/concurrency/main_executor.hpp

```diff
@@ -13,7 +13,12 @@
 
 #include <stlab/config.hpp>
 
-#if STLAB_MAIN_EXECUTOR(QT)
+#if STLAB_MAIN_EXECUTOR(QT5) || STLAB_MAIN_EXECUTOR(QT6)
+#include <QtGlobal>
+#if (STLAB_MAIN_EXECUTOR(QT5) && (QT_VERSION < QT_VERSION_CHECK(5,0,0) || QT_VERSION >= QT_VERSION_CHECK(6,0,0)) || \
+     STLAB_MAIN_EXECUTOR(QT6) && (QT_VERSION < QT_VERSION_CHECK(6,0,0) || QT_VERSION >= QT_VERSION_CHECK(7,0,0)))
+#error "Mismatching Qt versions"
```

Comment:
Interestingly, mismatch will also be caught by cmake itself early on: if a target links (transitively) with both Qt5 and Qt6, it will get conflicting definitions for the property `QT_MAJOR_VERSION` :not_bad:

> CMake Error: The INTERFACE_QT_MAJOR_VERSION property of "Qt5::Core" does
not agree with the value of QT_MAJOR_VERSION already determined
for "main".

Reply:
Cool